redash/handlers/users.py

```diff
@@ -27,7 +27,7 @@ def get(self):
     def post(self):
         req = request.get_json(force=True)
         require_fields(req, ('name', 'email'))
-
+        req['email'] = req['email'].lower()
```

Comment:
@ShahSunny I agree about one central change, but this one you did is not the central one.

@wichert's suggestion to use a case insensitive type for this column makes sense, although need to think how to deal with the migration - considering that some users might have the same email more than once already.